src/release.js

```diff
@@ -0,0 +1,62 @@
+const fs = require('fs');
+const util = require('util');
+
+const readFile = util.promisify(fs.readFile);
+const exists = util.promisify(fs.exists);
+
+const {validateChangelog} = require('./validate');
+
+const validateRelease = async (changelog, branch) => {
+    console.info(changelog);
+    const folder = changelog.replace('CHANGELOG.md', '');
+
+    const changelogContent = await readFile(changelog, {encoding: 'utf-8'});
+
+    const {
+        isUnreleased,
+        version,
+        date,
+        skeleton,
+    } = validateChangelog(changelogContent);
+
+    if (isUnreleased) {
+        throw new Error(`"${branch}" branch can't be unreleased`);
+    }
+
+    if (!version || version === 'Unreleased') {
+        throw new Error(`"${branch}" branch should have a version`);
+    }
+
+    if (!date) {
+        throw new Error(`"${branch}" branch should have a date`);
+    }
+
+    const {version: packageVersion} = JSON.parse(await readFile(`${folder}package.json`, {encoding: 'utf-8'}));
+    if (packageVersion !== version) {
+        throw new Error(`The package version "${packageVersion}" does not match the newest version "${version}"`);
```

Comment:
please always give project name context in error messages

Reply:
I opted to always show the changelog that is evaluating, this way we always know the project that is evaluating.